src/analysis/locate.ml

```diff
@@ -310,34 +312,36 @@ let rec load_cmt comp_unit ml_or_mli =
       (* there might not have been an mli (so no cmti), so the decl comes from
           the .ml, and the corresponding .cmt *)
       log ~title:"load" "Failed to load cmti file, retrying with cmt";
-      load_cmt comp_unit `ML
+      load_cmt ~config comp_unit `ML
     end else
       Error ()
 
-module Shape_reduce =
-  Shape.Make_reduce (struct
-    type env = Env.t
-
-    let fuel = 10
-
-    let read_unit_shape ~unit_name =
-      let fn = File.(with_ext (cmt unit_name)) in
-      log ~title:"read_unit_shape" "inspecting %s" unit_name;
-      match Load_path.find_uncap fn with
-      | filename ->
-        let cmt_infos = (Cmt_cache.read filename).cmt_infos in
-        move_to filename cmt_infos;
-        log ~title:"read_unit_shape" "shapes loaded for %s" unit_name;
-        cmt_infos.cmt_impl_shape
-      | exception Not_found ->
-        log ~title:"read_unit_shape" "failed to find %s" fn;
-        None
 
-    let find_shape env id = Env.shape_of_path
-      ~namespace:Shape.Sig_component_kind.Module env (Pident id)
-  end)
 
-let uid_of_path ~env ~ml_or_mli ~decl_uid path ns =
+let uid_of_path ~config ~env ~ml_or_mli ~decl_uid path ns =
+  let module Shape_reduce = Shape.Make_reduce (struct
```

Comment:
What's the impact of making this module local w.r.t. caching?

Reply:
I would say none since the cache is not shared between different calls to `reduce`

Reply:
I would say none since the cache is not shared between different calls to `reduce`

Reply:
test duplication